hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -1351,6 +1357,33 @@ private class ResponseParams {
 
     @Override
     public String toString() {
+      boolean isCallerContextEnabled = conf.getBoolean(
+          HADOOP_CALLER_CONTEXT_ENABLED_KEY,
+          HADOOP_CALLER_CONTEXT_ENABLED_DEFAULT);
+      CallerContext context = getCallerContext();
+      if (isCallerContextEnabled && context != null && context.isContextValid()) {
+        String cc = context.getContext();
+        String clientIp = "";
+        String ipKey = CallerContext.CLIENT_IP_STR +
```

Comment:
HDFS-13248 use `isProxyUsers` to control whether obtain client's ip and port from `CallerContext`. But in this PR, you plan ignore `isProxyUsers`? 

And where is the toString method used in NameNode?

Reply:
I don't think there is a conflict with HDFS-13248 here.
For Router, it will always pass Clientip and port to CallerContext, but NameNode may not always use it, which is not a problem with isProxyUsers.
Logging the Clientip in the log file would look clearer.